src/command-parser/expand-npm-shortcut.ts

```diff
@@ -2,21 +2,38 @@ import { CommandInfo } from '../command';
 import { CommandParser } from './command-parser';
 
 /**
- * Expands commands prefixed with `node:`, `npm:`, `yarn:`, `pnpm:`, or `bun:` into the full version `npm run <command>` and so on.
+ * Expands shortcuts according to the following table:
+ *
+ * | Syntax          | Expands to            |
+ * | --------------- | --------------------- |
+ * | `npm:<script>`  | `npm run <script>`    |
+ * | `pnpm:<script>` | `pnpm run <script>`   |
+ * | `yarn:<script>` | `yarn run <script>`   |
+ * | `bun:<script>`  | `bun run <script>`    |
+ * | `node:<script>` | `node --run <script>` |
+ * | `deno:<script>` | `deno task <script>`  |
  */
 export class ExpandNpmShortcut implements CommandParser {
     parse(commandInfo: CommandInfo) {
-        const [, npmCmd, cmdName, args] =
-            commandInfo.command.match(/^(node|npm|yarn|pnpm|bun):(\S+)(.*)/) || [];
-        if (!cmdName) {
+        const [, prefix, script, args] =
+            /^(npm|yarn|pnpm|bun|node|deno):(\S+)(.*)/.exec(commandInfo.command) || [];
```

Comment:
Out of curiosity, was there any advantage of changing from `string.match` to `regexp.exec()` here?
AFAICT both are generally equivalent, but maybe I'm missing something.

No dramas if it's just your preference :)

Reply:
Mostly personal preference, yeah. From what I've heard, `String.match()` and `RegExp.exec()` function identically to each other if the global flag is not set, in fact, `String.match()` uses `RegExp.exec()` under the hood, so using the latter carries a very minor performance benefit with it.